### PR TITLE
Use a popup theme to style the picker on Android

### PR DESCRIPTION
**Pull request by redstar504**
cc @shawnborton for style approval

### Details

This PR introduces a popup theme for styling the picker background on Android.  After some recent findings, I believe we should reopen the styling discussion.  

In the issue thread, we agreed on using a theme that more closely matches the modals.  As part of that discussion @shawnborton mentioned using a border. After further testing, I have identified a minor problem related to borders on a popup theme.  When scrolling, the items overlap the border.  Unfortunately this is not fixable through styling, but would require modifying the Picker source directly and adding `ViewGroup.clipToPadding(true)` which is not feasible.  

This problem can be identified in the following video:

<details>
<summary>Border overlap issue</summary>

#### 
These examples are from a demo app where I could quickly iterate changes, which is why the items are not ones used in our app.  This demonstration of the overlap issue uses a 6dp border:

https://user-images.githubusercontent.com/1311325/226196349-bec0b1a5-38b3-48b4-a975-1b19c098e771.mp4

</details>

There are a couple ways we could approach this. Since the modals in the app themselves do not use a border, I think we could get away without them. I think it looks great with just the font, card background color, and border radius, and it also solves the issue:

<details>

<summary>Popup theme without border</summary>

em_no_border

p6_no_border

</details>

Another idea would be to use a thinner border which would make the overlap less obvious.  Here are a couple examples of how it would look with narrower border widths:

<details>

<summary>Popup theme (2dp border)</summary>

em_border_2dp

p6_border_2dp

</details>

<details>
<summary>Popup theme (4dp border)</summary>

em_border_4dp

p6_border_4dp

</details>

Since having no border solves the issue and looks really good, I have included that version in this PR.  If @shawnborton decides we should include the border despite the overlap issue, I can quickly add a commit for that.

### Fixed Issues

$ https://github.com/Expensify/App/issues/15460
PROPOSAL: https://github.com/Expensify/App/issues/15460#issuecomment-1444500449

#### Picker Test

1. Launch the Android native app
2. Log in and navigate to a screen containing a picker
3. Open the picker and confirm that the picker has a green background
4. Confirm the picker has a border radius
5. Confirm the items are displayed with a white text color, in the Expensify Neue font.
6. Suggested: Run the app on a physical device to confirm the same functionality.

Example pickers used for testing:
- Workspace > General Settings > Default currency
- Workspace > Reimburse Expenses > Track Distance Unit
- Workspace > Personal Details > Home Address > Country

#### DatePicker Test

1. Launch the Android native app
2. Log in and navigate to a screen containing a DatePicker
3. Open the datepicker and confirm the style matches the Android native screenshots in the PR.
7. Suggested: Run the app on a physical device to confirm the same functionality.

Date picker used for testing:
- Workspace > Connect Bank Account > Connect Manually > Step 2 > Incorporation Date

#### Alert Test

1. Launch the Android native app
2. Log in and navigate to a screen that triggers an alert
3. Trigger the alert and confirm the style matches the Android native screenshots in the PR.
7. Suggested: Run the app on a physical device to confirm the same functionality.

Alerts used for testing:
- Profile > Upload Profile Image > Deny Camera Access
- Workspace > Upload Workspace Image > Deny Camera Access
- Report > Download Attachment > Success

- [X] Verify that no errors appear in the JS console

### QA Steps

- [X] Verify that no errors appear in the JS console

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [X] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

[Screencast from 2023-03-19 11:30:39 AM.webm](https://user-images.githubusercontent.com/1311325/226199050-33f867cd-10af-4756-9104-3a183bb6f294.webm)

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/1311325/226198823-2f7f3c70-5604-497d-b928-0d49e8fafb63.mp4

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/1311325/226198838-4a2491db-5dcd-4171-be25-30b30541ab80.mp4

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/1311325/226198847-216ca91a-310e-458f-9cba-dfd1dadb9ddd.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/1311325/226198858-9a3ecdad-2302-4dd8-8063-399d151a9cb6.mov

</details>

<details>
<summary>Android</summary>

#### Picker

em_picker

p6_picker

#### DatePicker

em_date

p6_date

#### Alert

em_alert

p6_alert

</details>

### CODE REVIEW

**Review by sobitneupane — COMMENTED** [review]
@redstar504 Let's update the background color for DatePicker and Alerts as well.

**Comment by shawnborton** [comment]
I think it looks pretty good without the border, so that works for me.

**Comment by shawnborton** [comment]
Do we have control over the color used to highlight the day, as well as "Cancel" below? If so, let's use green.

**Comment by redstar504** [comment]
@shawnborton Does this work?

<details>

<summary>Screenshot</summary>

datepicker-newcolor

</details>

**Comment by redstar504** [comment]
Commit for the recommended accent color is pushed.

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Android</summary>

- Workspace > General Settings > Default currency



- Workspace > Reimburse Expenses > Track Distance Unit



- Settings > Personal Details > Home Address > Country



- Add BankAccount > Company Information (step2) > State



- Add BankAccount > Company Information (step2) > Company Type



- DatePicker



- Alert



</details>

<details>
<summary>Web</summary>
- Datepicker
Screenshot 2023-03-21 at 12 52 56

- Dropdown
Screenshot 2023-03-21 at 12 54 49

</details>

<details>
<summary>Mobile Web - Chrome</summary>

- Datepicker
Screenshot 2023-03-21 at 13 26 50

- Dropdown
Screenshot 2023-03-21 at 13 26 44
</details>

<details>
<summary>Mobile Web - Safari</summary>

- Datepicker


- Dropdown


</details>

<details>
<summary>Desktop</summary>

- Datepicker
Screenshot 2023-03-21 at 13 26 50

- Dropdown
Screenshot 2023-03-21 at 13 26 44

</details>

<details>
<summary>iOS</summary>
- Datepicker


- Dropdown


</details>

File: android/app/src/main/res/drawable/datepicker_background.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+    <solid android:color="@color/card_alternate_bg" />
+    <corners android:radius="@dimen/dialog_radius" />
+</shape>
```
Comment (sobitneupane) [inline android/app/src/main/res/drawable/datepicker_background.xml:6]: @redstar504 Can you please add new line at the end of file.

File: android/app/src/main/res/drawable/picker_background.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+    <solid android:color="@color/card_highlight_bg" />
+    <corners android:radius="@dimen/dialog_radius" />
+</shape>
```
Comment (sobitneupane) [inline android/app/src/main/res/drawable/picker_background.xml:6]: @redstar504 Can you please add new line at the end of file.

File: android/app/src/main/res/values/dimen.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <dimen name="dialog_radius">16dp</dimen>
+    <dimen name="spinner_font_size">16dp</dimen>
+</resources>
```
Comment (sobitneupane) [inline android/app/src/main/res/values/dimen.xml:5]: @redstar504 Can you please add new line at the end of file.

**Review by sobitneupane — COMMENTED** [review]
@redstar504 Please do include DatePicker and Alert in Test Steps as well..

**Comment by situchan** [comment]
One more case:
image

**Comment by redstar504** [comment]
@sobitneupane Requested changes completed.

@situchan That case has been accounted for.

Screenshot from 2023-03-21 07-19-11

IMG-0686

**Comment by situchan** [comment]
@redstar504 great! I meant to add in test steps 🙂

**Review by sobitneupane — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

**Review by chiragsalian — APPROVED** [review]

**Comment by mvtglobally** [comment]
@redstar504 @jsamr  we need to QA this?

**Pull request merged**